File: MaWPictures/src/main/java/us/mikeandwan/photos/ui/photos/BasePhotoDialogFragment.java
package us.mikeandwan.photos.ui.photos;

import android.app.DialogFragment;
import android.content.Context;

import us.mikeandwan.photos.models.Photo;
import us.mikeandwan.photos.ui.HasComponent;


public class BasePhotoDialogFragment extends DialogFragment {
    protected <C> C getComponent(Class<C> componentType) {
        return componentType.cast(((HasComponent<C>)getActivity()).getComponent());
    }


    public Context getContext() {
        return getActivity().getBaseContext();
    }


    private IPhotoActivity getPhotoActivity() {
        return (IPhotoActivity) getActivity();
    }


    void updateProgress() {
        getPhotoActivity().updateProgress();
    }


    Photo getCurrentPhoto() {
        return getPhotoActivity().getCurrentPhoto();
    }
}

File: MaWPhotos/src/main/java/us/mikeandwan/photos/uiold/photos/BasePhotoDialogFragment.kt
package us.mikeandwan.photos.uiold.photos

import android.content.Context
import androidx.fragment.app.DialogFragment
import us.mikeandwan.photos.models.Photo

open class BasePhotoDialogFragment : DialogFragment() {
    override fun getContext(): Context {
        return requireActivity()
    }

    protected val photoActivity: IPhotoActivity
        get() = activity as IPhotoActivity

    fun addWork() {
        photoActivity.addWork()
    }

    fun removeWork() {
        photoActivity.removeWork()
    }

    val currentPhoto: Photo
        get() = photoActivity.currentPhoto!!
}